dbt_airflow_factory/builder.py

```diff
@@ -73,52 +105,110 @@ def _create_task_group_for_model(
+    def _create_task_for_model(
+        self,
+        model_name: str,
+        is_ephemeral_task: bool,
+        use_task_group: bool,
+        is_dbt_at_least_1: bool,
+    ) -> ModelExecutionTask:
+        if is_ephemeral_task:
+            return ModelExecutionTask(EphemeralOperator(task_id=f"{model_name}__ephemeral"), None)
+
         (task_group, task_group_ctx) = self._create_task_group_for_model(model_name, use_task_group)
         is_in_task_group = task_group is not None
         with task_group_ctx:
             run_task = self._make_dbt_run_task(model_name, is_in_task_group)
-            test_task = self._make_dbt_test_task(model_name, is_in_task_group)
+            test_task = self._make_dbt_test_task(model_name, is_in_task_group, is_dbt_at_least_1)
             # noinspection PyStatementEffect
             run_task >> test_task
         return ModelExecutionTask(run_task, test_task, task_group)
 
     def _create_tasks_for_each_model(
         self, manifest: dict, use_task_group: bool
-    ) -> Dict[str, ModelExecutionTask]:
+    ) -> Tuple[Dict[str, ModelExecutionTask], Dict[tuple, ModelExecutionTask]]:
         tasks = {}
+        tests_with_more_deps: Dict[tuple, List[str]] = {}
+        is_dbt_at_least_1 = self._get_dbt_version(manifest).major >= 1
+
         for node_name, node in manifest["nodes"].items():
+            node_name = cast(str, node_name)
             if self._is_model_run_task(node_name):
                 logging.info("Creating tasks for: " + node_name)
-                model_name = node_name.split(".")[-1]
-
-                tasks[node_name] = (
-                    ModelExecutionTask(EphemeralOperator(task_id=f"{model_name}__ephemeral"), None)
-                    if self._is_ephemeral_task(node)
-                    else self._create_task_for_model(model_name, use_task_group)
+                tasks[node_name] = self._create_task_for_model(
+                    node_name.split(".")[-1],
+                    self._is_ephemeral_task(node),
+                    use_task_group,
+                    is_dbt_at_least_1,
+                )
+            elif self._is_test_task(node_name):
+                model_dependencies = list(
+                    filter(self._is_model_run_task, node["depends_on"]["nodes"])
                 )
-        return tasks
+                if len(model_dependencies) <= 1:
+                    continue
+                model_dependencies.sort()
+                if tuple(model_dependencies) not in tests_with_more_deps:
+                    tests_with_more_deps[tuple(model_dependencies)] = []
+                tests_with_more_deps[tuple(model_dependencies)].append(node["name"])
+
+        multiple_deps_test_tasks = {}
+        for depends_on_tuple, test_names in tests_with_more_deps.items():
+            multiple_deps_test_tasks[depends_on_tuple] = ModelExecutionTask(
+                self._make_dbt_multiple_deps_test_task(test_names, depends_on_tuple),
+                None,
+            )
```

Comment:
This method became unreadable, can you refactor please? Maybe you can do runs and tests separately?